### DISCUSSION TITLE
Static file generator

### DISCUSSION THREAD

**Discussion by JoshuaBehrens** [code-hike/codehike#298] [Ideas and feedback]
<details>
<summary>
Scenario
</summary>

Hi I am looking to swap my code highlighting in our framework documentation to use codehike. We only use static files in our hosting and barely use javascript to keep our page performant and "pure". I am not afraid, that codehike is bloated, it seems not to be! But I try to keep the page in a similar state as it is right now.
</details>

I fiddled down a small PoC on how I can pre-render the mdx files in a CI/CD. It works out quite ok.

```main.js
import { evaluate } from '@mdx-js/mdx'
import * as provider from '@mdx-js/react'

import { remarkCodeHike } from "@code-hike/mdx";

import React from 'react'
import * as runtime from 'react/jsx-runtime';
import { renderToStaticMarkup } from 'react-dom/server'

import fs from 'fs';

const theme = JSON.parse(fs.readFileSync('node_modules/shiki/themes/nord.json'));
const mdxOptions = {
    remarkPlugins: [
        [
            remarkCodeHike,
            {
                theme,
                lineNumbers: false
            }
        ],
    ],
    useDynamicImport: true
};
const template = fs.readFileSync('demo.mdx', 'utf-8');
const { default: MDXContent } = await evaluate(
    template,
    {
        ...provider,
        ...runtime,
        ...mdxOptions
    }
);
const css = fs.readFileSync("node_modules/@code-hike/mdx/dist/index.css", 'utf-8');
const rendered = renderToStaticMarkup(React.createElement(MDXContent));
const page = `<style>${css}</style>${rendered}`;

await fs.writeFileSync('demo.html', page);
```

demo.html renders visually fine except:

<img width="384" alt="image" src="https://user-images.githubusercontent.com/1133593/200206904-5f2691e8-0ca7-4a37-bef6-901b6c40e48b.png">

* Code is hidden due to an opacity: 0
* tab switch can not display other file as its content is not in the HTML
* tab switch does not function

I am not surprised that static page is not interactive. I am neither surprised that due to the lack of a bundler I can't import json file nor include css by `import`. Some features can be patched manually but other demos will definitely not work that easily, like the scroll watcher code change.

Are there any plans to support static file generation? Is it already possible and I don't see it? I am fine with non-interactive solutions (as in printable) where I at least see a correctly highlighted code snippet.

**Comment by pomber** [comment]
Hi, you need JS to use Code Hike. No plans to make it work without JS at the moment. I'm experimenting with something like that in [another project](https://github.com/code-hike/bright), but it's too early.

**Comment by JoshuaBehrens** [comment]
I see the need for JS for the complete feature set as well. But highlighting code without interactive features would still be possible. Can one support `bright` in any way?
